prov/efa/src/rxr/rxr_cq.c

```diff
@@ -831,6 +831,9 @@ void rxr_cq_handle_tx_completion(struct rxr_ep *ep, struct rxr_tx_entry *tx_entr
 		}
 	}
 
+	if (tx_entry->tx_id < 0)
+		return;
```

Comment:
I missed this earlier: here you need to call rxr_release_tx_entry() before return.